server/lib/__init__.py

```diff
@@ -24,19 +25,22 @@
 from .http_provider import HTTPImportProvider
 from .import_providers import ImportProviders
 from .null_provider import NullImportProvider
-from .resolvers import DOIResolver, ResolutionException, Resolvers
+from .resolvers import DOIResolver, ResolutionException, Resolvers, MinidResolver
 from .zenodo.auth import ZenodoVerificator
 from .zenodo.provider import ZenodoImportProvider
 
 RESOLVERS = Resolvers()
 RESOLVERS.add(DOIResolver())
+RESOLVERS.add(MinidResolver())
 
 IMPORT_PROVIDERS = ImportProviders()
 IMPORT_PROVIDERS.addProvider(BDBagProvider())
+IMPORT_PROVIDERS.addProvider(DerivaProvider())
```

Comment:
Order here matters unfortunately. `BDBagProvider` just checks if url/pid ends with `.zip`, which is true for Deriva resolved urls, e.g.:

```
https://identifiers.fair-research.org/hdl:20.500.12633/Uzn848Mmg1Wk ->
https://pbcconsortium.s3.amazonaws.com/wholetale/7c066384317e1395b28e082c66209a25/2021-12-01_13.38.24/Dataset_1-882P.zip
```

For now, I'd suggest pushing DerivaProvider before BDBagProvider until we figure out better solution.

Reply:
Yes, indeed! Will fix.

Reply:
Should be fixed. Not sure why github is not marking this as outdated.